src/crimson/osd/osd_operation.cc

```diff
@@ -177,7 +176,11 @@ seastar::future<> OperationThrottler::acquire_throttle(
 {
   crimson::osd::scheduler::item_t item{params, seastar::promise<>()};
   auto fut = item.wake.get_future();
+  bool was_empty = scheduler->empty();
   scheduler->enqueue(std::move(item));
+  if (was_empty) {
+    wake();
+  }
```

Comment:
This fix is unrelated to the in_progress change, but seems correct.  Calling wake() unconditionally would also work.

Reply:
It might make sense to pull this out into its own PR.  Generally, try not mix unrelated fixes into the same commit.